src/rt/runtime.rs

```diff
@@ -0,0 +1,449 @@
+use std::cell::Cell;
+use std::io;
+use std::iter;
+use std::ptr;
+use std::sync::atomic::{self, AtomicBool, Ordering};
+use std::sync::{Arc, Mutex};
+use std::thread;
+use std::time::Duration;
+
+use crossbeam_deque::{Injector, Steal, Stealer, Worker};
+use crossbeam_utils::thread::scope;
+use once_cell::unsync::OnceCell;
+
+use crate::rt::Reactor;
+use crate::task::Runnable;
+use crate::utils::{abort_on_panic, random, Spinlock};
+
+thread_local! {
+    /// A reference to the current machine, if the current thread runs tasks.
+    static MACHINE: OnceCell<Arc<Machine>> = OnceCell::new();
+
+    /// This flag is set to true whenever `task::yield_now()` is invoked.
+    static YIELD_NOW: Cell<bool> = Cell::new(false);
+}
+
+/// Scheduler state.
```

Comment:
I would love a brief overview of terminology here. Words like "machine" and "processor" mean specific things in this context, and are necessary to understand to make sense of what's going on.

It seems like a "machine" is somewhere between a thread and a task. Is this like the old blocking pool? And a "processor" exists as one per core, so I assume it's like just the general worker queue?